src/compile/selection/index.ts

```diff
@@ -0,0 +1,113 @@
+import {Binding, NewSignal, SignalRef} from 'vega';
+import {stringValue} from 'vega-util';
+import {FACET_CHANNELS} from '../../channel';
+import {
+  BrushConfig,
+  SELECTION_ID,
+  SelectionInit,
+  SelectionInitArray,
+  SelectionResolution,
+  SelectionType
+} from '../../selection';
+import {accessPathWithDatum, Dict} from '../../util';
+import {EventStream} from '../../vega.schema';
+import {FacetModel} from '../facet';
+import {isFacetModel, Model} from '../model';
+import {UnitModel} from '../unit';
+import interval from './interval';
+import multi from './multi';
+import single from './single';
+import {SelectionProjection, SelectionProjectionComponent} from './transforms/project';
+
+export const STORE = '_store';
+export const TUPLE = '_tuple';
+export const MODIFY = '_modify';
+export const SELECTION_DOMAIN = '_selection_domain_';
+export const VL_SELECTION_RESOLVE = 'vlSelectionResolve';
+
+export interface SelectionComponent<T extends SelectionType = SelectionType> {
+  name: string;
+  type: T;
+
+  // Use conditional typing (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html)
+  // so we have stricter type of init (as the type of init depends on selection type)
+  init?: (T extends 'interval'
```

Comment:
Why is `init` an array here? It seems like it should only be a `SelectionInitArray`, `SelectionInit`, or `SelectionInit | SelectionInit[]`, no?